### PR TITLE
MMDP-334 Enable search and filtering of stakeholders

### PR DESCRIPTION
**Pull request by fgkinus**
**What does this PR do?**

- add functionality for searching under the Stakeholders Directory section
- add functionality for filtering search results by State or Local Government Area
- add modal for viewing stakeholder details

**Description of Task to be completed?**

As an administrator, I want to be able to search stakeholders displayed in the 'Stakeholders Directory' section of the 'Resources' page by entering search criteria in a search box.I should also be able to filter the selection by state by applying filters.

**How should this be manually tested?**

```
git clone https://github.com/andela/mmdp_site
cd mmdp_site
git checkout ft-search-filter-stakeholders-MMDP-334
yarn run: server
yarn client:dev
```
- browse to `<localhost>:<port>/resources/stakeholders`

**Any background context you want to provide?**

To add stakeholders, reference the pull request for adding new stakeholders : 
https://github.com/andela/mmdp_site/pull/47

Furthermore, this Implementation requires mongo transactions to be enabled. Mongoose requires mongo version >4.0.0 and replica sets to be enabled. In line with this please refer to this guide for setting up the development environment and also more about replica sets.

about mongo transactions: https://mongoosejs.com/docs/transactions.html
setup-guide : http://thecodebarbarian.com/introducing-run-rs-zero-config-mongodb-runner.html

**What are the relevant Jira issues?**

[MMDP-334](https://viisaus.atlassian.net/browse/MMDP-334)

**Screenshots (if appropriate)**
- list stakeholders
image

- modal for viewing stakeholder details
image

- results filtered by state
image

- search by string
image

- filter by LGA
image

### CODE REVIEW

**Comment by p8ul** [comment]
Good work @fgkinus. Do you mind fixing the linting issues.
screenshot 2019-03-08 at 09 36 39

File: server/routes/api/stakeholdersDirectory.js [hunk 1]
```diff
@@ -118,6 +119,7 @@ export const update = async (req, res) => {
 
 export const list = async (req, res) => {
   try {
+    // const stakeholderDirectory =
```
Comment (p8ul) [inline server/routes/api/stakeholdersDirectory.js]: Please, don't commit commented out code

**Comment by fgkinus** [comment]
> Good work @fgkinus. Do you mind fixing the linting issues.
> screenshot 2019-03-08 at 09 36 39

I have fixed this and removed the unused imports

**Comment by fgkinus** [comment]
> Do you mind center aligning this loader.
> screenshot 2019-03-08 at 10 15 21

I have fixed this and increased the width of the loader
image

File: server/routes/api/stakeholdersDirectory.js (hunk 1 above)
Comment (fgkinus) [reply server/routes/api/stakeholdersDirectory.js]: I have removed the commented code

**Pull request closed without merging**